### DISCUSSION TITLE
HasAuthentication mechanism does not work properly

### DISCUSSION THREAD

**Discussion by AndrzejHuzandrow** [selenide/selenide#2431] [Q&A]
Hello, 
I use Selenide (6.15 -> 6,17,1) to test sites behind BasicAuth. Everything went well until HasAuthentication mechanism was introduced. 
With 6.15 there was no problem with basic auth. I used 
open(String relativeOrAbsoluteUrl, AuthenticationType authenticationType, Credentials credentials) overloaded method
with empty "" domain. But with current version the Authorization header is added to the requests. And it is, indeed, However chrome determines such requests as not safe (http), thereby a lot of downloaded fonts are not decoded. I can not figure out what is a reason. 
Any help will be appreciated.

Thanks

**Comment by asolntsev** [comment]
@AndrzejHuzandrow Instead of empty domain "", you need to set only domains that really need BasicAuth. Apparently, fonts don't need it.

**Reply by AndrzejHuzandrow** [reply]
Thanks for suggestion, unfortunately  specific domain did not solve the problem. Moreover I unwrapped plain selenium driver and register basic credentials directly with. It does not help either. 
But when I used fiddler as a proxy to add request header **Authorization: Basic ddddddd** everything worked fine. Fonts were downloaded and decoded without any problem

**Reply by asolntsev** [reply]
What exactly mean "did not solve the problem"? 
Put a breakpoint into method `com.codeborne.selenide.drivercommands.BasicAuthUtils#hostMatchesDomain`, and you will see for which requests authorization header is being sent.

**Reply by AndrzejHuzandrow** [reply]
It appeared that according to W3C specs chromedriver does not add Authorization header while fetching fonts. And our nginx controller has improper configuration having font path behind basic auth. As for domain name, Authorization headers are provided only  for specified domain. All correct here.
Thanks for explanation!

**Reply by asolntsev** [reply]
Wow. I didn't even know about such a spec. :)

**Reply by AndrzejHuzandrow** [reply]
Yes, it sounds wired but is true. I ran same test on geckodriver, as it was stated that FF has no such specification, and all the fonts were fetched without any problem
